hotfix/v0.17.6

astrochun opened the pull request:
<!-- IMPORTANT: Please do not create a PR without creating an issue first. -->

<!-- Fields in **bold** are REQUIRED, fields in *italics* are OPTIONAL. -->

**Description**
This address a bug with parsing preferred citation. See #180 for details and examples.

<!-- Add any linked issue(s) -->
Fixes #180


**Update Changelog**

- [x] README.md, [changelog](../../README.md#changelog) <!-- update changelog here -->


**Bump version**

v0.17.5 -> v0.17.6

- [x] README.md, [installation instructions](../../README.md#installation-instructions)
- [x] [`setup.py`](../../setup.py)
- [x] [`ldcoolp/__init__.py`](../../ldcoolp/__init__.py)


*Screenshots or additional context*
<!-- Add any other context about the problem here and/or screenshots to help explain the problem. -->


*Testing (if applicable)*
<!-- Explain how you tested this bug fix so that others can replicate it. -->
<!-- Example: The exact commands you ran and their output. -->

astrochun reviewed (commented):
These changes look good and have been manually tested to cover two edge cases:
1. Period for middle initial on all authors, specifically last author
2. Also avoids `et al.` for long author list that the citation is shortened to only eight

damian-romero commented on ldcoolp/curation/inspection/readme/__init__.py:
```diff
@@ -217,11 +218,12 @@ def retrieve_article_metadata(self):
-                    single_str_citation.replace('):', ').').split('. ')]
-        citation_list = [content for content in str_list[0:-2]]
-        citation_list.append(f"{str_list[-2]} {str_list[-1]}")
+        END_SENT = re.compile('((?<=[.!?])|(?<=\.\":)) +(?=[A-Z,0-9])')
+        citation_list = list(
+            filter(None, END_SENT.split(single_str_citation.replace('):', ').')))
+        )
```
The regexp seems solid. I am not sure If only having straight quotation marks `"` may be enough of it you would like to include curvy `“`  quotation marks as well as in `(?<=\.\["“]:))` .

One suggestion is making things more explicit and documenting the string processing to improve readability and maintainability:

```suggestion
        # Replace colon with '.' if preceded by ')'
        pre_processed_str = single_str_citation.replace('):', ').')
        # Match spaces following group 1, only if followed by group 2
        END_SENT = re.compile('((?<=[.!?])|(?<=\.\":)) +(?=[A-Z,0-9])')
        citation_list = []
        for section in END_SENT.split(pre_processed_str):
            if section:
                citation_list.append(match)
```

astrochun replied:
I'm OK with additional documentation and breaking it apart, but some of the docs above this block is already present. For example, it says: "This forces a period after the year and ensures multiple rows with the last two row[s] merged for simplicity". So some succinctness is necessary.

astrochun replied:
There is actually a typo. `match` is not defined.

damian-romero reviewed (commented):
The code looks good. I only had one comment that I think might be worth considering.

astrochun commented:
> The code looks good. I only had one comment that I think might be worth considering.

I like many of your changes, but it would have introduced a bug so I implemented it and made some minor changes. I still like using `filter` as it keep code concise. Explained the use of `filter` in the context of `re.split` outputs.

astrochun reviewed (commented):
Changes look good and has been tested:

Outputed result:
```
Preferred citation (DataCite format):

Hamilton, Christopher W.; Scheidt, Stephen P. (2019).
2015 Unmanned aerial system (UAS)-derived orthoimage mosaics and digital terrain models of the NE distal lava flow margin, Holuhraun lava flow-field, Vatnajökull National Park, Iceland.
University of Arizona Research Data Repository.
Dataset. https://doi.org/10.25422/azu.data.13269452
```
